contracts/factory/src/contract.rs

```diff
@@ -275,6 +277,57 @@ impl FactoryTrait for Factory {
             .get(&DataKey::Config)
             .expect("Factory: No multihop present in storage")
     }
+
+    fn get_user_portfolio(env: Env, sender: Address) -> UserPortfolio {
```

Comment:
```suggestion
    fn query_user_portfolio(env: Env, sender: Address) -> UserPortfolio {
```

Plus add a boolean flag `staking: Bool`; query the staking information only if it is true.

Reply:
alright

Reply:
Just `staking: bool`